packages/shared/friends/sagas.ts

```diff
@@ -435,6 +427,70 @@ export function getFriendRequests(request: GetFriendRequestsPayload) {
   getUnityInstance().AddFriendRequests(addFriendRequestsPayload)
 }
 
+export function getUnseenMessagesByUser() {
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  const conversations = client.getAllCurrentConversations()
+
+  const conversationsWithMessages = conversations.filter((conv) => conv.conversation.hasMessages)
+  const unseenPrivateMessages = {}
+
+  for (const conversation of conversationsWithMessages) {
+    unseenPrivateMessages[conversation.conversation.userIds![1]] = conversation.conversation.unreadMessages?.length || 0
+  }
+
+  const totalUnseenMessages = {
+    unseenPrivateMessages
+  }
+  getUnityInstance().UpdateTotalUnseenMessagesByUser(totalUnseenMessages)
+}
+
+export function getFriendsWithDirectMessages(request: GetFriendsWithDirectMessagesPayload) {
+  const client: SocialAPI | null = getSocialClient(store.getState())
+  if (!client) return
+
+  const conversations = client.getAllCurrentConversations()
+
+  const conversationsWithMessages = conversations.filter((conv) => conv.conversation.hasMessages)
```

Comment:
How about extracting this into a selector function as it seems it's a common pattern (also used in `getUnseenMessagesByUser`)?